R/readS4Nproject_input.R

```diff
@@ -1,90 +1,93 @@
+  regionmapping$Region[regionmapping$iso == "GRL"] <- "OECD Europe"
+  regionmapping$Region[regionmapping$iso == "IMN"] <- "OECD Europe"
+  regionmapping$Region[regionmapping$iso == "JEY"] <- "OECD Europe"
+  regionmapping$Region[regionmapping$iso == "MAF"] <- "Rest Centr America"
+  regionmapping$Region[regionmapping$iso == "MNE"] <- "Eastern Europe"
+
   # read in data
-  if (subtype=="co2prices") {
-    data <- read.csv("IMAGE_data_26112020_co2price.csv", stringsAsFactors=F)
-  } else if (subtype=="bioenergy") {
-    data <- read.csv("IMAGE_data_26112020_bioenergy.csv", stringsAsFactors=F)
+  if (subtype == "co2prices") {
+    data <- read.csv("IMAGE_data_26112020_co2price.csv", stringsAsFactors = FALSE)
+  } else if (subtype == "bioenergy") {
+    data <- read.csv("IMAGE_data_26112020_bioenergy.csv", stringsAsFactors = FALSE)
   } else {
     stop("Specify subtype to be read in: co2prices or bioenergy (total bioenergy demand)")
   }
-  
+
   # region name correction
-  data$Region[data$Region=="Russia+"]         <- "Russia +"
-  data$Region[data$Region=="China+"]          <- "China +"
-  data$Region[data$Region=="Rest C. America"] <- "Rest Centr America"
-  data$Region[data$Region=="Rest S. America"] <- "Rest South America"
-  data$Region[data$Region=="Rest S. Asia"]    <- "Rest South Asia"
-  data$Region[data$Region=="Rest S. Africa"]  <- "Rest Southern Africa"
-  
+  data$Region[data$Region == "Russia+"]         <- "Russia +"
+  data$Region[data$Region == "China+"]          <- "China +"
+  data$Region[data$Region == "Rest C. America"] <- "Rest Centr America"
+  data$Region[data$Region == "Rest S. America"] <- "Rest South America"
+  data$Region[data$Region == "Rest S. Asia"]    <- "Rest South Asia"
+  data$Region[data$Region == "Rest S. Africa"]  <- "Rest Southern Africa"
+
   data      <- merge(data, regionmapping, "Region")
-  data      <- data.frame(scenario=data$Scenario,year=data$Year,value=data$Value,iso=data$iso, stringsAsFactors=F)
-  data$year <- paste0("y",data$year)
+  data      <- data.frame(scenario = data$Scenario,
+                          year = data$Year,
+                          value = data$Value,
+                          iso = data$iso,
+                          stringsAsFactors = FALSE)
+  data$year <- paste0("y", data$year)
   x         <- collapseNames(as.magpie(data))
-  
-  if (subtype=="bioenergy"){
+
+  if (subtype == "bioenergy") {
     # Disaggregate regional bioenergy to country values using population weight
     # country population in SSP2
-    pop     <- calcOutput("Population", 
-                          naming = "indicator.scenario",
-                          extension2150 = "none",
-                          aggregate = FALSE, 
-                          years = paste0("y", c(2005, seq(2010, 2100, by = 10))), 
-                          round = 6)
-    pop     <- collapseNames(pop[,,"SSP2"])
+    pop     <- collapseNames(calcOutput("Population",
```

Comment:
You can drop "collapseNames" entirely :)

Reply:
Did you run the comparison with `scenario = "SSP2"`?

Reply:
Ok, I see... collapseNames deletes the data name, if only 1 data dimension exists.